src/inputController.ts

```diff
@@ -31,35 +31,40 @@ export class InputController {
                 const data: IFileCreatorInputData = {
                     inputName,
                     pathToCreateAt: this.directoryPathToCreateAt,
-                    templateFolderPath: this.templateFolderPath,
-                    templateName,
+                    templateFolderPath: templateDirectory,
                 };
                 return data;
             });
 
     }
 
-    private showTemplatePickerDialog(templateFolderPath: string): Promise<string> {
+    private showTemplatePickerDialog(templateFolderPath: string[]): Promise<string> {
         return new Promise((resolve, reject) => {
-
-            let templateNames: string[];
-            try {
-                templateNames = this.availableTemplateNames(templateFolderPath);
-            } catch (error) {
-                // tslint:disable-next-line:max-line-length
-                reject(new Error(`${constants.ERROR_SETUP_MESSAGE_PREFIX} Could not find folder: ${templateFolderPath}. Please see ${constants.README_URL} for information on setting up Blueprint in your project.`));
-                return;
+            let templates: string[] = [];
+
+            for (const templatePath of templateFolderPath){
+                let templateNames: string[];
+                try {
+                    templateNames = this.availableTemplateNames(templatePath);
+
+                    const templateObject: string[] = templateNames.map((str) => templatePath + "\\" + str);
+                    templates = templates.concat(templateObject);
+                } catch (error) {
+                    // return;
+                    // tslint:disable-next-line:max-line-length
+                    // return;
+                }
             }
 
-            if (templateNames.length === 0) {
+            if (templates.length === 0) {
                 // tslint:disable-next-line:max-line-length
-                reject(new Error(`${constants.ERROR_SETUP_MESSAGE_PREFIX} No templates found in: ${templateFolderPath}. Please see ${constants.README_URL} for information on setting up Blueprint in your project.`));
-                return;
+                reject(new Error(`${constants.ERROR_SETUP_MESSAGE_PREFIX} No templates found. Please see ${constants.README_URL} for information on setting up Blueprint in your project.`));
+                /// return;
```

Comment:
Let's return here so any side-effects after the reject don't run (that's my understanding of promises at least).